fastai/widgets/image_cleaner.py

```diff
@@ -94,117 +99,158 @@ def sort_idxs(cls, similarities):
-        if handler is not None: btn.on_click(handler)
+        btn.on_click(handler)
         if style is not None: btn.button_style = style
-        btn.file_path = file_path
-        btn.flagged_for_delete = False
+        if img_idx is not None: btn.img_idx = img_idx
         return btn
 
-    @classmethod
-    def make_dropdown_widget(cls, description='Description', options=['Label 1', 'Label 2'], value='Label 1',
-                            file_path=None, layout=Layout(), handler=None):
+    @staticmethod
+    def make_dropdown_widget(options:Collection, value, handler:Callable, img_idx:Optional[int]=None,
+                             description='', layout=Layout(width='auto')) -> widgets.Dropdown:
         "Return a Dropdown widget with specified `handler`."
         dd = widgets.Dropdown(description=description, options=options, value=value, layout=layout)
-        if file_path is not None: dd.file_path = file_path
-        if handler is not None: dd.observe(handler, names=['value'])
+        dd.observe(handler, names='value')
+        if img_idx is not None: dd.img_idx = img_idx
         return dd
 
-    @classmethod
-    def make_horizontal_box(cls, children, layout=Layout()):
+    @staticmethod
+    def make_horizontal_box(children:Collection[widgets.Widget], layout=Layout()) -> widgets.HBox:
         "Make a horizontal box with `children` and `layout`."
         return widgets.HBox(children, layout=layout)
 
-    @classmethod
-    def make_vertical_box(cls, children, layout=Layout(), duplicates=False):
+    @staticmethod
+    def make_vertical_box(children:Collection[widgets.Widget],
+                          layout=Layout(width='auto', height='300px', overflow_x="hidden")) -> widgets.VBox:
         "Make a vertical box with `children` and `layout`."
-        if not duplicates: return widgets.VBox(children, layout=layout)
-        else: return widgets.VBox([children[0], children[2]], layout=layout)
+        return widgets.VBox(children, layout=layout)
 
-    def create_image_list(self, dataset, fns_idxs):
+    def create_image_list(self, fns_idxs:Collection[int], drop_batch_on_nonfile=False) -> Iterator[ImgData]:
         "Create a list of images, filenames and labels but first removing files that are not supposed to be displayed."
-        items = dataset.x.items
-        if self._duplicates:
-            chunked_idxs = chunks(fns_idxs, 2)
-            chunked_idxs = [chunk for chunk in chunked_idxs if Path(items[chunk[0]]).is_file() and Path(items[chunk[1]]).is_file()]
-            return  [(dataset.x[i]._repr_jpeg_(), items[i], self._labels[dataset.y[i].data]) for chunk in chunked_idxs for i in chunk]
-        else:
-            return [(dataset.x[i]._repr_jpeg_(), items[i], self._labels[dataset.y[i].data]) for i in fns_idxs if
-                    Path(items[i]).is_file()]
+        items = self._dataset.x.items
+        idxs = ((i for i in fns_idxs if items[i].is_file())
+                if not drop_batch_on_nonfile
```

Comment:
If `ImageCleaner` is split into two: cleaner and duplicate finder, `create_image_list` in the base class can be simplified by removing `drop_batch_on_nonfile`. Duplicate finder widget will simply override `create_image_list` instead of passing a constructor argument.